Default status / NO DATA docs (#14691)
DataDog/documentation: content/en/monitors/create/configuration.md

{{< tabs >}}
{{% tab "Metric based monitors" %}}

`Do not notify` if data is missing or `Notify` if data is missing for more than `N` minutes.

You either get notified for data being missing, or not. The notification occurs when no data was received during a configured time window.

**Note**: It is recommended that you set the missing data window to at least two times the evaluation period.

If you are monitoring a metric over an auto-scaling group of hosts that stop and start automatically, notifying for no data would produce a lot of notifications. For this case, you should not enable notifications for missing data. This option does not work if it is enabled at a time when data has not been reporting for a long period.

##### Grouping

For a monitor that does not notify on missing data, if a group does not report data, the monitor skips evaluations and eventually drops the group. During this period, the bar in the results page stays green. When there is data and groups start reporting again, the green bar shows an OK status and backfills to make it look like there was no interruption.

{{% /tab %}}

{{% tab "Other monitor types" %}}

If data is missing for `N` minutes:

> - `Evaluate as zero` / `Show last known status`
> - `Show NO DATA`
> - `Show NO DATA and notify`
> - `Show OK`.
> 
> {{< img src="/monitors/create/on_missing_data.png" alt="No Data Options" style="width:70%;">}}

Reviewer: ```suggestion
{{< img src="/monitors/create/on_missing_data.png" alt="No Data Options" style="width:70%;">}}

- `Evaluate as zero` / `Show last known status`
- `Show NO DATA`
- `Show NO DATA and notify`
- `Show OK`.
```

The selected behaviour is applied once a monitor's query does not return any data. Hence, contrary to the `Do not notify` option, the missing data window is **not** configurable.

| Option                    | Monitor status & notification                                             |
|---------------------------|---------------------------------------------------------------------------|
| `Evaluate as zero`        | Empty result is "replaced" with zero and compared to the alert/warning thresholds. For example, if the alert threshold is set to `> 10`, a zero would not trigger that condition, hence the Monitor status is set to `OK`.   |
| `Show last known status`  | The last known status of the group/monitor is set.                        |
| `Show NO DATA`            | Monitor status is set to `NO DATA`.                                       |
| `Show NO DATA and notify` | Monitor status is set to `NO DATA` and a notification is sent out.        |
| `Show OK`                 | Monitor is resolved and status is set to `OK`.                            |

> The options `Evaluate as zero` and `Show last known status` are shown depending on the type of the query:

Reviewer: ```suggestion
The `Evaluate as zero` and `Show last known status` options are displayed based on the query type:
```

> **Evaluate as zero**
> This option is available for monitors using `Count` queries.
> 
> **Show last known status**
> On the other hand, `Show last known status` is available for monitors using any other query type than `Count`, for example `Gauge`, `Rate`, or `Distribution`.

Reviewer: ```suggestion
- **Evaluate as zero:** This option is available for monitors using `Count` queries.
- **Show last known status:** This option is available for monitors using any other query type than `Count`, for example `Gauge`, `Rate`, and `Distribution`.
```

> Both options are considered as the **default status** of the monitor.

Reviewer: ```suggestion
Both options are the default status of the monitor.
```

Author: Removing this sentence alltogether.

In most cases this setting is not useful because you only want an alert to resolve once it is actually fixed. So, in general, it makes sense to leave this as `[Never]` so alerts only resolve when the metric is above or below the set threshold.

#### Group retention time

> Drop the group from the monitor status after `N` hours of missing data (can be set up to 72 hours).

Reviewer: ```suggestion
You can drop the group from the monitor status after `N` hours of missing data. The maximum length of time is 72 hours).
```

{{< img src="/monitors/create/group_retention_time.png" alt="Group Retention Time Option" style="width:70%;">}}

> Similar to the [Auto-resolve option][4], the group retention works when data is no longer being submitted. This option controls how long the group is kept in the monitor's status once data stops reporting. By default, a group is kept for 24h in the status before it's dropped. The start time of the group retention as well as the Auto-resolve option are **identical**, namely as soon as the monitor query returns no data.

Reviewer: ```suggestion
Similar to the [Auto-resolve option][4], the group retention works when data is no longer being submitted. This option controls how long the group is kept in the monitor's status once data stops reporting. By default, a group keeps the status for 24 hours before it is dropped. The start time of the group retention and the Auto-resolve option are **identical** as soon as the monitor query returns no data.
```

> Typical use cases for defining a group retention time are:
> - you would like to drop the group immediately/shortly after data stops reporting
> - you would like to keep the group in the status for as long as you usually take for troubleshooting

Reviewer: ```suggestion
Some use cases to define a group retention time include:

- When you would like to drop the group immediately or shortly after data stops reporting
- When you would like to keep the group in the status for as long as you usually take for troubleshooting
```

[1]: /monitors/create/#monitor-types
[2]: /monitors/notify/variables/?tab=is_alert#tag-variables
[3]: /monitors/notify/#renotify
[4]: /monitors/create/configuration#auto-resolve

> [5]: /monitors/create/configuration/?tab=othermonitortypes

Reviewer: ```suggestion
[5]: /monitors/create/configuration/?tabs=thresholdalert#no-data
```
